File: 디자인패턴/Python 디자인 패턴/Template/DisplayArticleTemplate.py
from abc import*
from Article import*
class DisplayArticleTemplate(metaclass=ABCMeta):
    @abstractmethod
    def title(this)->None:pass
    @abstractmethod
    def content(this)->None:pass
    @abstractmethod
    def footer(this)->None:pass
    
    @abstractmethod
    def __init__(this, article:Article):
        this.article = article
    
    def display(this):
        this.title()
        this.content()
        this.footer()
    pass

File: 디자인패턴/java 디자인 패턴/Template/DisplayArticleTemplate.java
public abstract class DisplayArticleTemplate {
    protected Article article;
    protected abstract void title();
    protected abstract void content();
    protected abstract void footer();

    public DisplayArticleTemplate(Article article) {
        this.article = article;
    }

    public final void display() {
        title();
        content();
        footer();
    }
}